File: cffu-core/src/test/java/io/foldright/cffu/CffuStateTest.java
package io.foldright.cffu;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.condition.EnabledForJreRange;
import org.junit.jupiter.api.condition.JRE;

import java.util.concurrent.Future;

import static org.junit.jupiter.api.Assertions.assertEquals;

class CffuStateTest {
    @Test
    @EnabledForJreRange(min = JRE.JAVA_19)
    void toCffuState() {
        assertEquals(CffuState.RUNNING, CffuState.toCffuState(Future.State.RUNNING));
        assertEquals(CffuState.SUCCESS, CffuState.toCffuState(Future.State.SUCCESS));
        assertEquals(CffuState.FAILED, CffuState.toCffuState(Future.State.FAILED));
        assertEquals(CffuState.CANCELLED, CffuState.toCffuState(Future.State.CANCELLED));
    }

    @Test
    @EnabledForJreRange(min = JRE.JAVA_19)
    void toFutureState() {
        assertEquals(Future.State.RUNNING, CffuState.RUNNING.toFutureState());
        assertEquals(Future.State.SUCCESS, CffuState.SUCCESS.toFutureState());
        assertEquals(Future.State.FAILED, CffuState.FAILED.toFutureState());
        assertEquals(Future.State.CANCELLED, CffuState.CANCELLED.toFutureState());
    }
}

File: cffu-core/src/test/java/io/foldright/cffu/CffuStateTest.kt
package io.foldright.cffu

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.condition.EnabledForJreRange
import org.junit.jupiter.api.condition.JRE.JAVA_19
import java.util.concurrent.Future

@EnabledForJreRange(min = JAVA_19)
private class CffuStateTest {
    @Test
    fun toCffuState() {
        CffuState.toCffuState(Future.State.RUNNING) shouldBe CffuState.RUNNING
        CffuState.toCffuState(Future.State.SUCCESS) shouldBe CffuState.SUCCESS
        CffuState.toCffuState(Future.State.FAILED) shouldBe CffuState.FAILED
        CffuState.toCffuState(Future.State.CANCELLED) shouldBe CffuState.CANCELLED
    }

    @Test
    fun toFutureState() {
        CffuState.RUNNING.toFutureState() shouldBe Future.State.RUNNING
        CffuState.SUCCESS.toFutureState() shouldBe Future.State.SUCCESS
        CffuState.FAILED.toFutureState() shouldBe Future.State.FAILED
        CffuState.CANCELLED.toFutureState() shouldBe Future.State.CANCELLED
    }
}
